ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/CommandParser.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.tools.migrations.util;
+
+import io.confluent.ksql.api.client.ColumnType.Type;
+import io.confluent.ksql.execution.expression.tree.BooleanLiteral;
+import io.confluent.ksql.execution.expression.tree.CreateArrayExpression;
+import io.confluent.ksql.execution.expression.tree.CreateMapExpression;
+import io.confluent.ksql.execution.expression.tree.CreateStructExpression;
+import io.confluent.ksql.execution.expression.tree.DecimalLiteral;
+import io.confluent.ksql.execution.expression.tree.DoubleLiteral;
+import io.confluent.ksql.execution.expression.tree.Expression;
+import io.confluent.ksql.execution.expression.tree.IntegerLiteral;
+import io.confluent.ksql.execution.expression.tree.LongLiteral;
+import io.confluent.ksql.execution.expression.tree.StringLiteral;
+import io.confluent.ksql.metastore.TypeRegistry;
+import io.confluent.ksql.parser.AstBuilder;
+import io.confluent.ksql.parser.DefaultKsqlParser;
+import io.confluent.ksql.parser.KsqlParser;
+import io.confluent.ksql.parser.tree.InsertValues;
+import io.confluent.ksql.tools.migrations.MigrationException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.ws.rs.NotSupportedException;
+
+public class CommandParser {
+  private static final Pattern STRING_WS_SEMICOLON =
+      Pattern.compile("('([^']*|(''))*')|;", Pattern.DOTALL);
+  private static final Pattern INSERT_VALUES_PATTERN = Pattern.compile(
```

Comment:
Does this work if there are appearances of parens or the string `VALUES` within quotes within the statement? I think it should be fine but regexes are notoriously hard to reason about. Might be nice to add quick unit tests just for this pattern to cover some of these edge cases.

Reply:
This regex doesn't look like it works with quoted stream names (both backticks and double quotes are supported). I think the more reliable thing to do is to chop up the statement to remove quoted strings (similar to the logic above) and then check for the tokens (minus whitespace) `INSERT` and `INTO` at the front and `VALUES` appearing anywhere else in the statement, rather than relying on a single regex.